modules/schema/src/main/java/org/apache/ignite/internal/schema/RowAssembler.java

```diff
@@ -338,6 +340,17 @@ public void appendBitmask(BitSet bitSet) {
      * @return Serialized row.
      */
     public byte[] build() {
+        if (schema.keyColumns() == curCols)
+            throw new IllegalStateException("Key column missed: colIdx=" + curCol);
+        else {
+            if (curCol == 0)
+                flags |= BinaryRow.RowFlags.NULL_VALUE;
+            else if (schema.valueColumns().length() != curCol)
+            throw new IllegalStateException("Value column missed: colIdx=" + curCol);
```

Comment:
Does it make sense to change `IllegalStateException` to a new `IgniteInternalException`?

Reply:
Well, it seems that we need IGNITE-14495 to be merged.